### PR TITLE
Add support for SwiftPM

### PR DESCRIPTION
**Pull request by mohssenfathi**
## Description
This PR adds support for Swift Package Manager as the default option for integrating the SDK. 

### Folder structure
By default, SPM expects sources to be under a top level Sources folder. This can be changed in the Package.swift, but I moved them to keep the default structure.

### Test location
Ideally, we would keep all the unit tests under the default top level Tests folder, however there are a couple issues with this:
1. There are several tests that rely on a functional keychain to pass. The keychain requires app entitlements which cannot be included without an app target to attach to. So these tests must be run from an app target's test cases.
2. SPM requires that all resources be accessed via `Bundle.module` instead of the `Bundle.main` and `Bundle(for: ...)` instances that were already being used. While technically feasible, this required a lot of changes.

### Removal of TestAppShim
TestAppShim was an empty target contained in the sources folder. Since the tests have been moved to the example app this target is no longer needed.
**Alternative**: We can move the test cases to TestAppShim and leave them out of the SwiftSDK sample app.

### Dropped Objc Support
Objc interfaces were dropped in this PR and all the corresponding `@objc` declarations (except for #selector) were removed. Likewise, the Objc SDK sample app was deleted.

### Manually importing UIKit + Foundation
The umbrella header defining global imports was removed from UberRides, so `import Foundation` and `import UIKit` needed to be manually added to every class that requires it.

### Unit tests
All unit tests should pass from inside the sample app target Swift SDK.
**Note**: I found a bug in some of the tests that will fail if the Uber Rides app is installed on the simulator. I'll fix this in a later PR.

### Manual testing
You can run the SwiftSDK sample app under `Examples/Swift SDK`. The behavior should remain the same.

### CODE REVIEW

File: Sources/UberAuth/UberAuth.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//
+//  Copyright © Uber Technologies, Inc. All rights reserved.
+//
+
+
+import Foundation
+
+final class UberAuth {}
```
Comment (chiragramani) [inline Sources/UberAuth/UberAuth.swift:8]: Where are we using this class?

File: Sources/UberCore/Authentication/Authenticators/BaseAuthenticator.swift [hunk 2]
```diff
@@ -25,27 +25,26 @@
...
 
 /// Base class for authorization flows
-@objc(UBSDKBaseAuthenticator) public class BaseAuthenticator: NSObject, UberAuthenticating {
+public class BaseAuthenticator: UberAuthenticating {
     /// Scopes to request during login
-    @objc public var scopes: [UberScope]
+    public var scopes: [UberScope]
     
-    @objc public var requestUri: String?
+    public var requestUri: String?
```
Comment (chiragramani) [inline Sources/UberCore/Authentication/Authenticators/BaseAuthenticator.swift:32]: How about having them as constant properties, assuming that we wouldn't want them to mutate once the instance is created?

File: Sources/UberCore/Configuration.swift [hunk 3]
```diff
@@ -183,7 +181,7 @@ private let callbackURIStringKey = "URIString"
     }
     
     /// The current version of the SDK as a string
-    @objc public var sdkVersion: String {
+    public var sdkVersion: String {
         guard let version = Bundle(for: Configuration.self).object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String else {
             return "Unknown"
         }
```
Comment (chiragramani) [inline Sources/UberCore/Configuration.swift:187]: Can it be a lazy var if this would be called more than once?

**Review by chiragramani — APPROVED** [review]

**Comment by mohssenfathi** [comment]
Discussed comments offline with @chiragramani

**Pull request merged**